### PR TITLE
[TM-1099] update site status

### PR DESCRIPTION
**Pull request by LimberHope**
[Task Link](https://gfw.atlassian.net/browse/TM-1099)

### CODE REVIEW

File: app/StateMachines/SiteStatusStateMachine.php [hunk 1]
```diff
@@ -12,6 +12,7 @@ public function transitions(): array
 
         $parentTransitions[self::NEEDS_MORE_INFORMATION][] = self::RESTORATION_IN_PROGRESS;
         $parentTransitions[self::APPROVED][] = self::RESTORATION_IN_PROGRESS;
+        $parentTransitions[self::AWAITING_APPROVAL][] = self::RESTORATION_IN_PROGRESS;
```
Comment (roguenet) [inline app/StateMachines/SiteStatusStateMachine.php:15]: If a site skips straight from awaiting approval to restoration in progress, the hooks in the state machine for `approved` won't go off. See `afterTransitionHooks` in the `EntityStatusStateMachine`. Perhaps what would be better would be to transition it to approved and then to restoration in progress to make sure the transition hooks go off as expected.

Comment (pachonjcl) [reply app/StateMachines/SiteStatusStateMachine.php:15]: image
My understanding is that restoration in progress is an intermediate state between awaiting approval and approved.

Comment (pachonjcl) [reply app/StateMachines/SiteStatusStateMachine.php:15]: So there is no need for firing off approved events.

Comment (roguenet) [reply app/StateMachines/SiteStatusStateMachine.php:15]: Oh, in that case is there any need to keep `approved` -> `restoration-in-progress`?

Comment (pachonjcl) [reply app/StateMachines/SiteStatusStateMachine.php:15]: It is exactly the rule that we need to address the new polygons creations that will immediately change a approved site to a "restoration-in-progress" site.

**Review by roguenet — APPROVED** [review]

**Pull request merged**